### ISSUE TITLE
[no-undef] Const enums considered as not defined

### ISSUE THREAD

**Issue by aliksend** [typescript-eslint/typescript-eslint#1330]
**Repro**

```JSON
{
  "rules": {
    "no-undef": ["error"]
  }
}
```

index.ts
```TS
/// <reference path='./types.d.ts'/>

const a = Lang.UK
console.log(a)
```

types.d.ts
```TS
declare const enum Lang {
    UK = "UK",
    RU = "RU",
    EN = "EN"
}
```

**Expected Result**

No errors

**Actual Result**

Message:
```
index.ts:3:11: 'Lang' is not defined.
```

**Additional Info**

Of course this code is working.
If I use `enum` instead of `const enum` code will fail in runtime with `ReferenceError: Lang is not defined` so this rule make sense, but only for non-const enum.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.11.0` |
| `@typescript-eslint/parser`        | `2.11.0` |
| `TypeScript`                       | `3.7.3` |
| `ESLint`                           | `6.7.2` |
| `node`                             | `10.16.3` |
| `npm`                              | `6.9.0`|

**Comment by bradzacher** [comment]
This isn't a problem with const enums, it's a problem with ambient definitions.
You would get a similar error if instead your `types.d.ts` contained:

```ts
declare const Lang: { UK: string };
```

It's not possible to build this check into the rule either, because ESLint operats on a single file at a time.

Closing this as we will not be supporting ref comments in the rule.

---

We recommend not using `no-undef` anyway, because the typescript compiler has the checks built in.
